linkerd/meshtls/rustls/src/creds/store.rs

```diff
@@ -1,37 +1,37 @@
-use super::params::*;
-use super::InvalidKey;
+use super::{params::*, InvalidKey};
 use linkerd_dns_name as dns;
 use linkerd_error::Result;
 use linkerd_identity as id;
 use linkerd_meshtls_verifier as verifier;
 use ring::{rand, signature::EcdsaKeyPair};
 use std::{convert::TryFrom, sync::Arc};
 use tokio::sync::watch;
-use tokio_rustls::rustls;
+use tokio_rustls::rustls::{self, pki_types::UnixTime, server::WebPkiClientVerifier};
 use tracing::debug;
 
 pub struct Store {
     roots: rustls::RootCertStore,
-    server_cert_verifier: Arc<dyn rustls::client::ServerCertVerifier>,
+    server_cert_verifier: Arc<dyn rustls::client::danger::ServerCertVerifier>,
     server_id: id::Id,
     server_name: dns::Name,
     client_tx: watch::Sender<Arc<rustls::ClientConfig>>,
     server_tx: watch::Sender<Arc<rustls::ServerConfig>>,
     random: ring::rand::SystemRandom,
 }
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 struct Key(Arc<EcdsaKeyPair>);
 
-#[derive(Clone)]
+#[derive(Clone, Debug)]
 struct CertResolver(Arc<rustls::sign::CertifiedKey>);
 
 pub(super) fn client_config_builder(
-    cert_verifier: Arc<dyn rustls::client::ServerCertVerifier>,
+    cert_verifier: Arc<dyn rustls::client::danger::ServerCertVerifier>,
 ) -> rustls::ConfigBuilder<rustls::ClientConfig, rustls::client::WantsClientCert> {
-    rustls::ClientConfig::builder()
-        .with_cipher_suites(TLS_SUPPORTED_CIPHERSUITES)
-        .with_safe_default_kx_groups()
+    let mut provider = rustls::crypto::ring::default_provider();
+    provider.cipher_suites = TLS_SUPPORTED_CIPHERSUITES.to_vec();
+
```

Comment:
It is probably preferable to provide a single `provider()` function that returns a provider with our customizations applied, decoupling most of our code from ring, ideally.

Reply:
Done